Bring PR up to date with `ethereum-optimism:superchain-registry/main`

geoknee opened the pull request:
Adds `optimismConfig` and activates a validation check for that.

tessr committed:
permit testnets and devnets to use newer contract versions (#508)

* permit testnets and devnets to use newer contract versions

* tidy

* don't perform bytecode checking for testnets

* avoid reimporting superchain package

refcell committed:
feat(bindings): Superchain Crate (#511)

* feat(bindings): superchain crate

* feat(bindings): superchain crate

refcell committed:
fix(bindings): Disable Default Features in superchain dep (#512)

* fix(bindings): superchain registry import

* fix: alias prefices

geoknee committed:
modify promotion test notification (#516)

geoknee committed:
Add silence for OP Mainnet Optimism_Portal_2_Params (#515)

* add silence for OP Mainnet Optimism_Portal_2_Params

* fix comment

* rework silences mechanism

* pull Granite time from superchain go module

* just lint-all

geoknee committed:
Store Optimism Config in config toml files (#510)

* capture optimism config when compressing genesis

* remove submodule

* move types down into superchain module

* lint

* pull in op-geth from https://github.com/ethereum-optimism/op-geth/pull/371

* migrate existing genesis files to include default Optimism values

EIP1559Elasticity: 6, EIP1559Denominator: 50, EIP1559DenominatorCanyon: u64ptr(250)

special case for base sepolia, overrides elasticity paramter to be 10

Our diff bot should pick up any mistakes in this migration, we are aiming for a net zero overall diff to the effective genesis files

* zora sepolia: set EIP1559DenominatorCanyon: nil

We have the actual genesis of this chain on hand, so we know the correct value of this parameter

* update op-geth

* remove temporary workaround code

* break line for readability

* move Optimism config from genesis .json.gz to chain config toml files

* update op-geth

* fix add-chain e2e test

* update chain configs to include genesis.config.optimism field

* restore indentation

* fix value

* just codegen

* add genesis.config.optimism to devnets

* just codegen

* Update superchain/configs/sepolia/zora.toml

* fix: rust-bindings codegen

* un-nest Optimism config

* update op-geth

* update config files with new optimism structure

* just codegen

* remove duplicate data

* fixup op sepolia (missed it before)

* just codegen

---------

Co-authored-by: refcell <abigger87@gmail.com>

geoknee committed:
Remove submodule (#521)

* remove submodule

* add optimism-temporary to gitignore

refcell committed:
feat(bindings): Remove Rust Bindings (#517)

geoknee committed:
update op-geth version (#523)

to a commit on the main optimism branch

geoknee committed:
fix gas-token_test (#522)

tessr committed:
tweak add-chain instructions (#528)

geoknee committed:
Update README.md (#524)

Remove link to bindings

geoknee committed:
add button with link to just-promotion-test slack message (#527)

cody-wang-cb committed:
Update batcher proposer (#526)

* update batcher proposer

* ran codegen

---------

Co-authored-by: George Knee <georgeknee@googlemail.com>

bitwiseguy committed:
validation: add Optimism Config test (#536)

* validation: add Optimism Config test

* validation: use string const for test names

* validation: add explanatory comment for testOptimismConfig

geoknee committed:
Use `op-contracts/v1.3.0` ABI for pre fault proofs validation (#537)

* improve test failure output for testSuperchainConfig

* use lowercase getters for nonFaultProofs standard roles

this is the API for MCP release (1.3.0)

* Update validation/superchain-config_test.go

* just lint-all

sbvegan committed:
Update superchain.toml (#542)

This timestamp is for Wednesday, not Tuesday https://www.epochconverter.com/

geoknee committed:
[ Feature Branch ] Genesis allocs validation (#489)

* fix go.work version declaration

* add genesis predeploy test, validation metadata for chains 10,34443 and wire up

* undo accidental changes

* fix ci config

* add sepolia/worldcoin

* add worldchain

* some small fixes

chain 480 passes!

* change format of genesis creation command

* add nvm

* 4801 passes

* introduce monorepoBuildCommand artifact

* try removing source command

* install nvm

* no need to install just

* linting

* introduce boolean switch for using legacy deployments format

* Get mode, metal and zora genesis.allocs validation working (#496)

* introduce node_version

need to reexamine writeDeploymentsLegacy

* fix bug in writeDeploymentsLegacy

* fix declaration of legacy deployments and add note

* improve diff

* use hardhat Deployment type to pad data and avoid unmarshaling errors

e.g. on mode

* fix writeDeploymentsLegacy, mode now validates!

* remove note

* introduce  GenesisCreationCommand mapping

allows metadata file to choose from a list and have chainId automatically injected into command

* introduce BuilldCommand mapping

* op sepolia validates

* remove UseLegacyDeploymentsFormat

* remove op mainnet metadata for now

* remove op sepolia (it doesn't validate yet)

* echo nvm version

* just codegen

* load nvm

* fix

* try installing nvm in a step

* require node_version to be set

* just lint-all

* attempt to patch genesis creation source code

 to make validation work at a wider range of commits

* adds metal metadata -- validates!

* add zora metadata -- validates!

* Enhance `add-chain` tool to capture deploy config and monorep commit information  (#504)

* add-chain: ingest and store deploy-config and genesis-creation commit

* remove genesis-system-configs dir

we no longer use this

* add test data for deploy-config

* check errs properly

* link validation to add-chain

* fix up add-chain e2e tests

* simplify test side effect handling

* don't bother cleaning up -test genesis validation inputs

(They now don't end up bound into the validation package)

* tweak spacing

* prefer os.ModePerm over 0o777

* typo

* typo

* remove unused params

* use commented constants for metadata defaults

* fix clean-add-chain cmd

* use localised, gitignored optimism-temporary directory for genesis validation purposes (#507)

* Get genesis validation checks to run on CI (#529)

* switch to cimg/go:1.22.6-node docker image for genesis validation

* remove set -e

(nvm.sh exits with code 3, we can apparently just ignore that)

* use bash instead of sh

* use . instead of source

* install pnpm

* move optimism-temporary out from under superchain-registry

this seems to make go module resolution work as expected

* do not allow validate-genesis to timeout

* improve pipefail behaviour

we allow the scripts which load nvm and gvm to fail

* just lint-all

* remove all set -e behaviour

gives highest chance that validation can succeed

* stream output from test

helps with debugging, prevents timeouts on circleci

* try running as a matrix

* add base and base sepolia to genesis validation list (#530)

* add base sepolia to genesis validation list

* add base mainnet to genesis validation list

* Add validation metadata for Lisk mainnet + testnet (#531)

* temp:  gk/genadd lisk mainnet (revertme)

* add lisk metadata for genesis validation

* temp: add lisk sepolia (revertme)

* add genesis validation metadata for lisk sepolia

* Update .circleci/config.yml

* add comment

* just tidy-all

* harmonize op-geth version

* improve naming of ci steps

* dedupe perChainTestName

* remove temporary directory after tests run

* tidy up test script

* rename test

* rename commands

* add testGenesisAllocsMetadata

* use dynamic config building in CI (#534)

* use dynamic config building in CI

This works by having a small workflow run, compute the list of chain ids to run genesis validation checks on, and then modify a subsequent config file which circleci then runs.

* add comment

* fan in

* fix

* fix script

* Update .circleci/continue_config.yml

Co-authored-by: Vinod Damle <vdamle@users.noreply.github.com>

---------

Co-authored-by: Vinod Damle <vdamle@users.noreply.github.com>

* remove temporarily added chains (#535)

* git restore --source=main -- superchain

* just codegen

* just lint-all

* lint

* typo

* enhance comment

* swap op-node1 and op-node2

to retain chronological ordering

* add explainer about parallelism

* modify message in genesis-allocs-all-ok

* simplify patch file

* tweak go version in golang-validate-genesis-allocs job

* remove the test artifacts written to validationInputsDir

* don't forget we swapped opnode1<>opnode2

* fix another opnode1<>opnode2 permutation

* Revert "simplify patch file"

This reverts commit 9b7f6d8bb00e417fd78e6a806390054402caaf08.

* fix op-node1/2 err

* add note about gvm

---------

Co-authored-by: Vinod Damle <vdamle@users.noreply.github.com>

bitwiseguy committed:
circleci: use context to pass rpc urls (#544)

* circleci: use context to pass rpc urls

* circleci: add rpc context to hourly/nightly jobs

geoknee committed:
Adds validation metadata for chain with id 9897 (arena-z / nod) (#540)



* add arena z testnet

* remove temporary files (fixmeup)

* update genesis_creation_commit

chose the one immediately before the l1StartingBlockTime

* upgrade validation code to cope with forge command

* just lint-all

* make go_version inference work even when in a subdir with no go.mod file

* run commands from the correct dirs

* fix typo

* git restore --source=main --staged --worktree superchain

* just codegen

* Update superchain/configs/mainnet/superchain.toml

---------

Co-authored-by: Vinod Damle <vdamle@users.noreply.github.com>

geoknee committed:
Modify validation data for Lisk (#543)

* modify deploy-config validation input metadata for lisk mainnet

* run git submodule update during test

* trigger genesis validation recheck when deploy-config file changes

refcell committed:
feat: Json Configs (#545)

geoknee committed:
add genesis validation metadata for mode sepolia (#546)

* add genesis validation metadata for mode sepolia

* update mode sepolia deploy-config

geoknee committed:
add validation metadata for zora sepolia (#547)

geoknee committed:
Merge remote-tracking branch 'origin/main' into lattice-redstone

geoknee committed:
add optimismConfig && just codegen

vdamle committed:
CI: Migrate scheduled workflows to scheduled pipeline jobs (#549)

* CircleCI has two pipeline triggers named `nightly_build` and `hourly_build`.
* Remove workflow triggers and configure jobs to run on scheduled pipeline
  triggers .

Co-authored-by: Vinod Damle <vinod@oplabs.co>

vdamle committed:
Genesis alloc utility: increase buffer size for streaming output/errors (#551)

default scan buffer size in bufio is 64k, which limits the max size of any token
that can be scanned. Since diffs for the genesis alloc tool has to process large hex strings,
we supply a larger buffer to ensure that output/errors are piped in its entirety.

Co-authored-by: Vinod Damle <vinod@oplabs.co>

geoknee committed:
Move `L1SecurityConfigs` and `SuperchainConfig` tests from standard candidate to standard only category (#552)

* bring subtest interface in line with others

* move l1 securityconfigs test to standard-only

remove exclusions

* remove unecessary exclusions

* tidy up the way we apply exclusions

This is now handled in a single, central location. I introduced typing for subtests to make this very clear how to extend in future.

* just lint-all

* use type aliasing

* revert to usual Go fn declarations

* fix some stragglers

geoknee committed:
Merge remote-tracking branch 'origin/main' into gk/lattice-redstone

tchardin commented on superchain/configs/mainnet/redstone.toml:
```diff
@@ -13,6 +13,11 @@ seq_window_size = 3600
 max_sequencer_drift = 600
 data_availability_type = "eth-da"
 
+[optimism]
```
Not sure where this section comes from but these values are incorrect, you should be able to get the correct one from our deploy config file

geoknee replied:
I reran the `just add-chain` command and these values were updated [467e8d8](https://github.com/latticexyz/superchain-registry/pull/1/commits/467e8d837da5f61e666b47dfe091c173bc623461)

tchardin commented on superchain/configs/mainnet/redstone.toml:
```diff
@@ -13,6 +13,11 @@ seq_window_size = 3600
 max_sequencer_drift = 600
 data_availability_type = "eth-da"
```
This is still wrong as well

geoknee replied:
Thanks for highlighting.

I think this is because of the renaming breaking change -- did you migrate your `rollup.json` yet?
https://docs.optimism.io/builders/chain-operators/features/alt-da-mode#modify-rollupjson-config

I have filed a new issue https://github.com/ethereum-optimism/superchain-registry/issues/557 so we can do a better job communicating this.

geoknee replied:
I migrated the rollup.json and reran the add-chain command here [467e8d8](https://github.com/latticexyz/superchain-registry/pull/1/commits/467e8d837da5f61e666b47dfe091c173bc623461)

geoknee committed:
modify optimism params for redstone

geoknee committed:
migrate redstone's rollup.json and rerun just add-chain

migrated to new "alt-da" scheme

https://docs.optimism.io/builders/chain-operators/features/alt-da-mode#modify-rollupjson-config
